### PR TITLE
[xbuild] TaskLoggingHelper: Implement support for correctly using tas…

### PR DESCRIPTION
**Pull request by radical**
…k resources

- make task resources available to TaskLoggingHelper
- verify TaskLoggingHelper method args
- tests

### CODE REVIEW

File: mcs/class/Microsoft.Build.Tasks/Test/Microsoft.Build.Tasks/TestMessageLogger.cs [hunk 1]
```diff
@@ -197,6 +197,29 @@ public int CheckAny (string text, MessageImportance importance)
...
+		public int CheckFullLog (string text)
+		{
+			if (all_messages.Count <= 0)
+				return 1;
+
+			int foundAt = -1;
+			for (int i = 0; i < all_messages.Count; i ++) {
+				BuildEventArgs arg = all_messages [i];
+				if (text == arg.Message) {
+					foundAt = i;
```
Comment (marek-safar) [inline mcs/class/Microsoft.Build.Tasks/Test/Microsoft.Build.Tasks/TestMessageLogger.cs]: Why not to do `all_messages.RemoveAt (i); return 0;` instead

Comment (radical) [inline mcs/class/Microsoft.Build.Tasks/Test/Microsoft.Build.Tasks/TestMessageLogger.cs]: Fixed.

File: mcs/class/Microsoft.Build.Tasks/Test/Microsoft.Build.Tasks/TestMessageLogger.cs [hunk 2]
```diff
@@ -197,6 +197,22 @@ public int CheckAny (string text, MessageImportance importance)
 			return 0;
 		}
 
+		public int CheckFullLog (string text)
+		{
+			if (all_messages.Count <= 0)
```
Comment (marek-safar) [inline mcs/class/Microsoft.Build.Tasks/Test/Microsoft.Build.Tasks/TestMessageLogger.cs]: This extra check is not needed

**Comment by radical** [comment]
Updated based on the feedback, thanks.

**Pull request merged**